robotoff/prediction/ocr/origin.py

```diff
@@ -0,0 +1,374 @@
+            "en": EN_ORIGIN_OCR_REGEX,
+        }
+        self.GENERAL_WORDS_BY_LANG: Dict[str, List[str]] = {
+            "fr": GENERAL_WORDS_FR,
+            "en": GENERAL_WORDS_EN,
+        }
+        self.COUNTRY_ADJECTIVE_TO_COUNTRY_NAME_BY_LANG = {
+            "fr": COUNTRY_ADJECTIVE_FR_TO_COUNTRY_NAME_FR,
+            "en": COUNTRY_ADJECTIVE_EN_TO_COUNTRY_NAME_EN,
+        }
+
+        self.OUTSIDE_WORDS_BY_LANG = {"fr": OUTSIDE_WORDS_FR, "en": OUTSIDE_WORDS_EN}
+
+    def find_origin(self, content: Union[OCRResult, str]) -> List[Prediction]:
+        """
+        Returns a list of Prediction, each prediction as a "data" field that contains a "ingredients_origins"
+        field. Each "ingredients_origins" dict has three fields :
+        "origin" -> the country, a region from which specified ingredients comes.
+        "same_for_all_ingredients" -> a boolean, if True then all the ingredient of the product have the
+        specified origin, and the third field "concerned_ingredient" is None
+        "concerned_ingredient" -> if "same_for_all_ingredients" is False, then this field is a list of the
+        ingredient that comes from the regions specified in "origin".
+        """
+        if not self.initialized:
+            self.initialize()
+
+        ingredients_origins: Dict[str, Dict[str, Any]] = {}
+        for lang, regex in self.EVERY_REGEX_BY_LANG.items():
+            text = get_text(content, regex)
+
+            if not text:
+                return []
+
+            for match in regex.regex.finditer(text):
+                origin = self.extract_origin_from_match(match, lang)
+                
+                if not origin or origin == OriginParser.UNKNOW_ORIGIN:
+                    continue
+
+                if not origin in ingredients_origins:
+                    ingredients_origins[origin] = {
+                        "origin": origin,
+                        "same_for_all_ingredients": True,  # True unless group "ingredients" matched
+                        "concerned_ingredients": None,
+                    }
+
+                if self.check_if_general_word_in_ingredients(
+                    self.GENERAL_WORDS_BY_LANG[lang], match.group("ingredients")
+                ):
+                    continue
+
+                if not match.group("ingredients"):
+                    continue
+
+                ingredients = self.extract_ingredients_from_match_string(
+                    match.group("ingredients"), lang
+                )
+                if ingredients is not None:
+                    ingredients_origins[origin]["same_for_all_ingredients"] = False
+                    ingredients_origins[origin]["concerned_ingredients"] = []
+                    for ingredient in ingredients:
+                        ingredients_origins[origin]["concerned_ingredients"].append(
+                            ingredient
+                        )
+
+        if len(ingredients_origins) == 0:
```

Comment:
These lines are not necessary

Reply:
You're right, it was usefull in old commits and is now useless